### DISCUSSION TITLE
auth, stripe and docs

### DISCUSSION THREAD

**Discussion by mronoffon** [lorinkoz/django-pgschemas#80] [Q&A] [answered]
Hi Lorenzo
(thanks for your reply/offer!  ...was too late reply, at the time)

So, still trying to figure out a good design for third-party integrations like Stripe, AllAuth and a long list
of other 'shared' apps.  So...

- My first conundrum is Customer sign-in.  After the initial signup, how does a 'customer' sign-in to their domain
with only email and password using allauth and Stripe)?   In other words, how will system 'recognize' the user as
a Tenant at login?

- How 'heavy' should the Tenant be?  Do you recommend the use of FK inside Tenant?

- If your documentation 'misses' anything, shall I refer to django_tenants' docs?

Thanks in advance

**Comment by lorinkoz** [comment, marked as answer]
Hi @mronoffon. Some comments to your questions:

1. It will depend on your tenant-user relationship. If you have only one user per tenant, that's one design, if you have multiple users that can be associated with multiple tenants, that's another design, and finally, you could have multiple users locked inside the tenant they belong to. With additional information, I could provide some guidance here.
2. No strict recommendation here, you can have a fat tenant model (you will keep all tenant data in the public schema), or you could have a thin tenant model and then a one-to-one to a singleton model that will live inside the tenant. There are no strong reasons to pick one over another.
3. `django-pgschemas` was forked from `django-tenants` and the core should work similarly, but there are important conceptual differences here. We have static tenants here, the settings are different, the routing is also slighly different here, etc. I would say that perhaps the docs of the original project won't be too helpful. In any case, you can point what's missing here so we can work in making it better.

Hope it helps.
